styx-scheduler-service/src/test/java/com/spotify/styx/api/SchedulerResourceTest.java

```diff
@@ -305,6 +307,18 @@ public void testCreateWorkflowInvalidConfiguration() throws Exception {
     verify(workflowValidator).validateWorkflowConfiguration(FULL_DAILY_WORKFLOW.configuration());
   }
 
+  @Test
+  public void testCreateWorkflowFailedToInitialize() throws Exception {
+    doThrow(new WorkflowInitializationException(new Exception()))
+        .when(workflowChangeListener).accept(FULL_DAILY_WORKFLOW);
+    final Response<ByteString> r = serviceHelper.request(
+        "POST", BASE + "/workflows/styx",
+        serialize(FULL_DAILY_WORKFLOW.configuration())).toCompletableFuture().get();
+    verify(workflowValidator).validateWorkflowConfiguration(FULL_DAILY_WORKFLOW.configuration());
+    verify(workflowChangeListener).accept(FULL_DAILY_WORKFLOW);
+    assertThat(r, hasStatus(withCode(Status.BAD_REQUEST)));
```

Comment:
Is `WorkflowInitializationException` always a bad request?

Reply:
I would say so. It happens when we cannot parse the `offset`, `schedule`.